docs/how-to/internal/release.rst

```diff
@@ -0,0 +1,72 @@
+============================
+How to cut a Rubicon release
+============================
+
+The release infrastructure for Rubicon is semi-automated, using Github
+Actions to formally publish releases.
+
+This guide assumes that you have an ``upstream`` remote configured on your
+local clone of the Rubicon repository, pointing at the official repository.
+It also assumes that you have a local ``release`` branch that tracks
+``upstream/master``.
+
+You should have a checkout of a personal fork of the Rubicon repository; to
+configure this fork for a release, run::
+
+    $ git remote add upstream git@github.com:beeware/rubicon-objc.git
```

Comment:
Good point; I'll clarify the language, with the assumption that an upstream exists.

Reply:
As for why SSH - If this were a set of instructions for general contributors, I'd agree; but this is internal documentation for maintainers. If you're a contributor to the repo in a position to make releases, I'd hope you're already using SSH (or can work out the details; or, at least, recognize there are two different URL schemes).

Reply:
So I've looked into this a little more, and it's not really clear to me what advantages you get by using Git over SSH rather than HTTPS.

First of all, GitHub themselves seem to prefer HTTPS Git. [This GitHub Help page](https://help.github.com/en/github/using-git/which-remote-url-should-i-use#cloning-with-https-urls-recommended) explicitly calls HTTPS "recommended", and the rest of the help pages and the web UI prefer HTTPS by default. [GitHub's docs for using SSH](https://help.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh) don't seem to explain anywhere why you'd want to use SSH rather than HTTPS.

The main advantage of SSH Git that I'm aware of is that you don't have to explicitly enter your password, and you can use different keypairs for each of your machines, so that you can selectively revoke them in case one gets compromised. However, GitHub lets you create [personal access tokens](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line), which allows basically the same thing with HTTPS Git.

From my personal experience, SSH is just a pain to set up correctly. You have to generate an SSH keypair, set up your SSH config to use that keypair when connecting to GitHub, make sure that your Git binary uses the right SSH binary, and add the public key to your GitHub account. If you get any of the steps wrong, there's no way to tell what the issue is. 
Compare that with HTTPS, where you only need to enter your username and password or an [access token](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line), and any decent Git frontend lets you save your credentials.

Sorry for the rant... I don't mind anyone using SSH Git - if it works for you, great! I just don't want to tell others to go through this complicated process if it doesn't have a proper benefit that anyone in the top DDG search results can articulate.

Reply:
Fair enough. If I had to guess, HTTPS is recommended because it's a darn sight easier to explain - "just type your password every time" - and walking someone through SSH credential creation is... not fun. 

For the purposes of documentation, I'm happy to move this back to HTTPS.

Reply:
> "just type your password every time"

OK, in that respect I'm clearly biased as a GUI user. With any competent Git GUI you can save your credentials with a checkbox, but setting up a command-line Git credential helper (especially if you want to use the macOS keychain) is about as fun as setting up SSH.

Also I just found out that [this exists](https://stackoverflow.com/a/36500841) - you can add some lines to your global Git config to automatically rewrite HTTPS URLs to SSH if that's what you prefer. Not relevant to this PR (frankly this entire discussion isn't...) but you might find it useful.